ast.ts

```diff
@@ -44,7 +45,7 @@ export type Literal =
   | { tag: "none" }
 
 // TODO: should we split up arithmetic ops from bool ops?
-export enum BinOp { Plus, Minus, Mul, IDiv, Mod, Eq, Neq, Lte, Gte, Lt, Gt, Is, And, Or};
+export enum BinOp { Plus, Minus, Mul, IDiv, Mod, Eq, Neq, Lte, Gte, Lt, Gt, Is, And, Or, Dot};
```

Comment:
I'm confused by this – how does `Dot` differ from field lookup? Is it syntactically any different?

Reply:
Reading the proposal, I believe this is being used to signify the matmul `@` operator in python. https://docs.python.org/3/library/operator.html#operator.matmul. The name is a little confusing though at first, esp since it's a bit more obscure syntax for vanilla python

Reply:
Yes. `Dot` is shortened for '"dot product of matrixes" or Numpy method `a.dot(b)` or `@` operator, which I thought is a very cool operator. Do you need me to change the enum to something like `Matmul`, `At`, etc.?

Reply:
Ooooohhh. Sorry I read the code and the proposal separately and didn't connect the.... dots.

Yeah let's call it MatMul, since that's the name of the dunder method. Thanks, sorry for the think-o on my part.